sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/PropagateEmptyRelation.scala

```diff
@@ -50,7 +50,7 @@ object PropagateEmptyRelation extends Rule[LogicalPlan] with PredicateHelper wit
   override def conf: SQLConf = SQLConf.get
 
   def apply(plan: LogicalPlan): LogicalPlan = plan transformUp {
-    case p @ Union(children) if children.exists(isEmptyLocalRelation) =>
+    case p @ Union(children, _, _) if children.exists(isEmptyLocalRelation) =>
```

Comment:
shall we skip if by-name resolution is not finished? Otherwise the final output schema can be indeterministic, if some children are empty relation and get removed before participating the by-name resolution.

Reply:
by-name resolution is finished after analysis. As PropagateEmptyRelation is run in optimization, I think by-name resolution should be done.

Reply:
You are right